src/app/static/src/tests/components/projects/projectHistory.test.ts

```diff
@@ -751,22 +767,22 @@ describe("Project history component", () => {
         const wrapper = getWrapper(testProjects);
         const vm = wrapper.vm as any
         const copyLink = wrapper.find("#v-s11").find(".version-cell.copy-cell").find("button");
-        copyLink.trigger("click");
-        await Vue.nextTick();
+        await copyLink.trigger("click");
+        await nextTick();
 
-        const modal = wrapper.findAll(".modal").at(1);
+        const modal = wrapper.findAll(".modal")[1];
         const input = modal.find("input");
-        const renameBtn = modal.find(".modal-footer").findAll("button").at(0);
+        const renameBtn = modal.find(".modal-footer").findAll("button")[0];
         await input.setValue("");
 
-        expect(renameBtn.attributes("disabled")).toBe("disabled");
+        expect(renameBtn.attributes("disabled")).toBe("");
```

Comment:
The "" vs undefined attribute values to check for disabled values is pretty confusing - might be more readable to use `(renameBtn.element as HTMLButtonElement).disabled` which gives you a boolean

Reply:
sounds like a job for regex!! will do this in a separate commit to your other changes as this may affect a fair number of files!

Reply:
..I'm pretty sure it lets you cast element in that case, done it lots in WODIN e.g. [here](https://github.com/mrc-ide/wodin/blob/798dedaabf6b4275dcf34207ead26461dfd482b3/app/static/tests/unit/components/run/runTab.test.ts#L116)
Maybe in all those cases the element type is included in the selector so it's smart enough to work out that "button#run-btn" will give you an HTMLButtonElement? So we could update the selectors in those cases - looks like there's only really a handful of examples.

Reply:
yep I believe the "button" part of the string is making all the difference actually, should be able to get this working!

Reply:
actually apologies! I was being dumb, you can cast it, I was forgetting the `.element` :disappointed: